src/psqlgraph/psql.py

```diff
@@ -1,74 +1,99 @@
+                host.
+            password: The password the driver should use for the given user.
+            database: The name of the database backing the graph represented in the
+                package namespace.
+            application_name: The name of this application by default will use the host
+                name. See connection_args.
+            auto_flush: Defaults to `True`; force all newly created sessions to set
+                autoflush. This value will be the default autoflush value and used while
+                creating new sessions. If the user passes a different value while
+                creating the session, this value will be ignored.
+            connection_args: The are the arguments which will be passed to
+                `sqlalchemy.create_engine`. Within this, the application name will be set
+                to the given value or its default.
+            isolation_level: Set the isolation_level kwarg of the
+                `sqlalchemy.create_engine` function. For more details, see its
+                documentation. Will be superseded by any value of the same name within
+                the kwargs.
+            package_namespace: The namespace for the graph that backs this driver. This
+                can be the default Edge/Node based graph provided in within `psqlgraph`
+                or a Edge/Node pair generated in the `psqlgraph.ext` module e.g. `bio`.
+            read_only: Defaults to `False`; controls whether new sessions are set to
+                only allow read only queries or not. This value is used while creating
+                new sessions and can be replaced by passing a different value while
+                creating the session.
+            set_flush_timestamps: Is `True` by default. Setting this to `True` will
+                perform an extra database query to get the server time at flush and
+                store `session._flush_timestamp`.
+            kwargs: Any additional parameters which should be passed to
+                `sqlalchemy.create_engine`. For details on what these are and their
+                usage see their official documentation.
         """
 
         # Parse kwargs
-        self.auto_flush = kwargs.pop("auto_flush", True)
-        self.read_only = kwargs.pop("read_only", False)
-        self.package_namespace = kwargs.pop("package_namespace", None)
-        connect_args = kwargs.pop("connect_args", {})
-        kwargs.pop("node_validator", None)
-        kwargs.pop("edge_validator", None)
-        self.set_flush_timestamps = kwargs.pop("set_flush_timestamps", True)
-        if "isolation_level" not in kwargs:
-            kwargs["isolation_level"] = "REPEATABLE_READ"
-        if "application_name" in kwargs:
-            connect_args["application_name"] = kwargs.pop("application_name")
-        else:
-            connect_args["application_name"] = socket.gethostname()
+        self.auto_flush = auto_flush
```

Comment:
Do you have any concerns about the downstream effects of not popping the values from kwargs since those get passed to the create_engine function?

Reply:
Actually this approach prevents that from happening. If you were to pass the arg application_name and then try and pass it as a kwarg python would compain:
```python
>>> def foo(x: int, **kwargs): ...
>>> foo(1, x=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() got multiple values for argument 'x'